### PR TITLE
Self managed public cloud

### PR DESCRIPTION
**Pull request by bmckercher123**
New "How to" as part of Get started reorg work.

- 3 docs for review: AWS, GCE, and Azure
- Written by Spike, reviewed by Barbara
- After review, merge ASAP

### CODE REVIEW

**Comment by caseydavenport** [comment]
Deploy preview for *calico* ready!

Built with commit d3115538dff607cdf75de2cc3c620f27a8a01586

https://deploy-preview-3156--calico.netlify.com

**Comment by caseydavenport** [comment]
Deploy preview for *calico* ready!

Built with commit 7bec21d09810cf2e16ee443ab3e516c7109d22df

https://deploy-preview-3156--calico.netlify.com

File: getting-started/kubernetes/self-managed-public-cloud/aws.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+1. [Set up your environment for AWS](https://kops.sigs.k8s.io/getting_started/aws/).
+1. Configure kops to use {{site.prodname}} for networking.  
+   The easiest way to do this is to pass `--networking calico` to kops when creating the cluster. For example
+
+   ```
+   kops create cluster \
+    --zones us-west-2a \
+    --networking calico \
+    name-of-your-cluster
+   ```
```
Comment (lmm) [inline getting-started/kubernetes/self-managed-public-cloud/aws.md]: This doesn't work, at least with kops `Version 1.15.0 (git-9992b4055)`. I get the following:

```
$    kops create cluster \
>     --zones us-west-2a \
>     --networking calico \
>     name-of-your-cluster

State Store: Required value: Please set the --state flag or export KOPS_STATE_STORE.
For example, a valid value follows the format s3://<bucket>.
You can find the supported stores in https://github.com/kubernetes/kops/blob/master/docs/state.md.
```

File: getting-started/kubernetes/self-managed-public-cloud/azu.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+#### Other options and tools
+
+##### {{site.prodname}} networking 
+
+You can also deploy {{site.prodname}} for both networking and policy enforcement. In this mode, {{site.prodname}} uses a VXLAN-based overlay network that masks the IP addresses of the pods from the underlying Azure VNET. This can be useful in large deployments or when running multiple clusters and IP address space is a big concern.
+
+Unfortunately, aks-engine does not support this mode, so you will have to use a different tool chain to install and manage the cluster. Some options:
+
+- Use [Terraform](https://docs.google.com/document/d/1CsOj3smZQHrmu0jbY7hg1ShXfcTswYm0Qj4BrFdw5qs/edit#heading=h.876rtqebbyno) to provision the Azure networks and VMs, then [kubeadm](https://docs.google.com/document/d/1CsOj3smZQHrmu0jbY7hg1ShXfcTswYm0Qj4BrFdw5qs/edit#heading=h.imdkqm7uo7st) to install the Kubernetes cluster.
```
Comment (lmm) [inline getting-started/kubernetes/self-managed-public-cloud/azu.md]: These look like internal links

File: getting-started/kubernetes/self-managed-public-cloud/azu.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+- Use [Kubespray](https://docs.google.com/document/d/1CsOj3smZQHrmu0jbY7hg1ShXfcTswYm0Qj4BrFdw5qs/edit#heading=h.1v8hzmk8q1x5)
```
Comment (lmm) [inline getting-started/kubernetes/self-managed-public-cloud/azu.md]: Same here, internal

File: getting-started/kubernetes/self-managed-public-cloud/gce.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+**kubeadm** is a cluster management tool that is used to install Kubernetes.
+
+### Before you begin...
+
+[Install and configure the Google Cloud CLI tools](https://cloud.google.com/sdk/docs/quickstarts)
+
+### How to
+
+There are many ways to install and manage Kubernetes in AWS. Using kubeadm is a good default choice for most people, as it gives you access to all of {{site.prodname}}’s [flexible and powerful networking features]({{site.baseurl}}/networking). However, there are other options that may work better for your environment.
```
Comment (lmm) [inline getting-started/kubernetes/self-managed-public-cloud/gce.md]: AWS => GCE/Google Cloud

File: getting-started/kubernetes/self-managed-public-cloud/gce.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+- [kubeadm for Calico networking and network policy](#kubeadm-for-calico-networking-and-network-policy)
+- [Other tools and options](#other-tools-and-options)
+
+#### kubeadm for Calico networking and network policy
+
+##### Create cloud resources
+
+You will need at least one VM to serve as a control plane node and one or more worker nodes. (It is possible to have control plane nodes also act as workers. This is not recommended in most cases and not covered by this guide.)  See [requirements]({{site.baseurl}}/getting-started/kubernetes/requirements) for specific OS requirements for these VMs.
+
+The following worked example creates a single control node and three workers on a dedicated virtual private network (VPC). Adjust the example as needed for your requirements. Consider a dedicated infrastructure management tool like [Terraform](https://docs.google.com/document/d/1-Vm8tdxc9GJ4JVXwVrHqQv96jU1eeadQeZyCDdU2bV4/edit#heading=h.876rtqebbyno) for managing cloud resources. (This example is adapted from [Kubernetes the Hard Way](https://github.com/kelseyhightower/kubernetes-the-hard-way/blob/master/docs/03-compute-resources.md).)
```
Comment (lmm) [inline getting-started/kubernetes/self-managed-public-cloud/gce.md]: These links are not public

File: getting-started/kubernetes/self-managed-public-cloud/gce.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+```
+kubectl apply -f calico.yaml
+```
+
+#### Other tools and options
+
+##### Amazon VPI CNI plugin
+
+An alternative to using {{site.prodname}} for both networking and network policy, is to use the Amazon’s VPC CNI plugin for networking and {{site.prodname}} for network policy. The advantage of this approach is that pods are assigned IP addresses associated with Elastic Network Interfaces on worker nodes. The IPs come from the VPC network pool and therefore do not require NAT to access resources outside the Kubernetes cluster.
```
Comment (lmm) [inline getting-started/kubernetes/self-managed-public-cloud/gce.md]: This is AWS-specific

File: getting-started/kubernetes/self-managed-public-cloud/gce.md (hunk 4 above)
Comment (bmckercher123) [reply getting-started/kubernetes/self-managed-public-cloud/gce.md]: Fixed.

File: getting-started/kubernetes/self-managed-public-cloud/aws.md (hunk 1 above)
Comment (spikecurtis) [reply getting-started/kubernetes/self-managed-public-cloud/aws.md]: @lmm You need to set up S3 and export the `KOPS_STATE_STORE` as described on: https://kops.sigs.k8s.io/getting_started/aws/

I've updated the second item above to hopefully make this a bit more clear. What do you think?

File: getting-started/kubernetes/self-managed-public-cloud/azu.md (hunk 2 above)
Comment (bmckercher123) [reply getting-started/kubernetes/self-managed-public-cloud/azu.md]: Fixed.

File: getting-started/kubernetes/self-managed-public-cloud/azu.md (hunk 3 above)
Comment (bmckercher123) [reply getting-started/kubernetes/self-managed-public-cloud/azu.md]: Fixed

File: getting-started/kubernetes/self-managed-public-cloud/gce.md (hunk 6 above)
Comment (bmckercher123) [reply getting-started/kubernetes/self-managed-public-cloud/gce.md]: ugh! fixed.

File: getting-started/kubernetes/self-managed-public-cloud/azu.md (hunk 3 above)
Comment (spikecurtis) [reply getting-started/kubernetes/self-managed-public-cloud/azu.md]: Not fixed! It still points to docs.google.com.

It's meant to point to the `Kubespray` heading on this page, which should be `[Kubespray](#kubespray)`, I think

File: getting-started/kubernetes/self-managed-public-cloud/azu.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+#### Other options and tools
+
+##### {{site.prodname}} networking 
+
+You can also deploy {{site.prodname}} for both networking and policy enforcement. In this mode, {{site.prodname}} uses a VXLAN-based overlay network that masks the IP addresses of the pods from the underlying Azure VNET. This can be useful in large deployments or when running multiple clusters and IP address space is a big concern.
+
+Unfortunately, aks-engine does not support this mode, so you must use a different tool chain to install and manage the cluster. Some options:
+
+- Use [Terraform](https://www.terraform.io/) to provision the Azure networks and VMs, then [kubeadm](https://docs.google.com/document/d/1CsOj3smZQHrmu0jbY7hg1ShXfcTswYm0Qj4BrFdw5qs/edit#heading=h.imdkqm7uo7st) to install the Kubernetes cluster.
```
Comment (spikecurtis) [inline getting-started/kubernetes/self-managed-public-cloud/azu.md]: kubeadm link is pointing to the original doc --- it's a cross reference.  Should be `[kubeadm](#kubeadm)` I think

File: getting-started/kubernetes/self-managed-public-cloud/azu.md (hunk 3 above)
Comment (bmckercher123) [reply getting-started/kubernetes/self-managed-public-cloud/azu.md]: Ah, I did not get that these were inline links. They were still pointing to wrong place though.

File: getting-started/kubernetes/self-managed-public-cloud/gce.md (hunk 5 above)
Comment (bmckercher123) [reply getting-started/kubernetes/self-managed-public-cloud/gce.md]: Fixed.

File: getting-started/kubernetes/self-managed-public-cloud/azu.md (hunk 7 above)
Comment (bmckercher123) [reply getting-started/kubernetes/self-managed-public-cloud/azu.md]: All inline links now.

File: getting-started/kubernetes/self-managed-public-cloud/aws.md (hunk 1 above)
Comment (lmm) [reply getting-started/kubernetes/self-managed-public-cloud/aws.md]: Looks good, thanks!

**Pull request merged**